src/rev_ai/generic_api_client.py

```diff
@@ -0,0 +1,167 @@
+                    delete_after_seconds=None,
+                    language=None,
+                    **kwargs):
+        """Submit a job to the api. This method is special in that it is intended to be hidden by
+        the implementation this is done because python standard is to pass options individually
+        instead of as an object and our true clients should match this standard
+
+        :param metadata: info to associate with the transcription job
+        :param callback_url: callback url to invoke on job completion as
+                             a webhook
+        :param delete_after_seconds: number of seconds after job completion when job is auto-deleted
+        :param language: specify language using the one of the supported ISO 639-1 (2-letter) or
+            ISO 639-3 (3-letter) language codes as defined in the API Reference
+        :returns: Job info object
+        :raises: HTTPError
+        """
+        payload = {}
+        for key, value in kwargs.items():
+            if value:
+                payload[key] = value
+
+        if metadata:
+            payload['metadata'] = metadata
+        if callback_url:
+            payload['callback_url'] = callback_url
+        if delete_after_seconds is not None:
+            payload['delete_after_seconds'] = delete_after_seconds
+        if language:
+            payload['language'] = language
+
+        response = self._make_http_request(
+            "POST",
+            urljoin(self.base_url, 'jobs'),
+            json=payload
+        )
+
+        return self.parse_job_info(response.json())
+
+    def get_job_details(self, id_):
+        """View information about a specific job.
+        The server will respond with the status and creation date.
+
+        :param id_: id of the job to be requested
+        :returns: Job info object
+        :raises: HTTPError
+        """
+        if not id_:
+            raise ValueError('id_ must be provided')
+
+        response = self._make_http_request(
+            "GET",
+            urljoin(self.base_url, 'jobs/{}'.format(id_))
+        )
+
+        return self.parse_job_info(response.json())
+
+    def get_list_of_jobs(self, limit=None, starting_after=None):
+        """Get a list of jobs submitted within the last 30 days in reverse
+        chronological order up to the provided limit number of jobs per call.
+        Pagination is supported via passing the last job id from previous call into starting_after.
+
+        :param limit: optional, limits the number of jobs returned,
+                      if none, a default of 100 jobs is returned, max limit if 1000
+        :param starting_after: optional, returns jobs created after the job with this id,
+                               exclusive (job with this id is not included)
+        :returns: list of jobs response data
+        :raises: HTTPError
+        """
+        params = []
+        if limit is not None:
+            params.append('limit={}'.format(limit))
```

Comment:
```suggestion
            params.append(f'limit={limit}')
```

Reply:
I think pretty much everywhere you use `.format` it should be replaced with this pattern

Reply:
fstrings are only available for python3+. We want to continue supporting python2 for the time being